src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,48 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import mate.academy.exception.NoUserInCacheException;
+import mate.academy.exception.RequestProcessingException;
 
 public class AsyncRequestProcessor {
+    private static Map<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(
+                () -> findUserInCache(userId), executor).exceptionally(ex -> {
+                    try {
+                        UserData userData = emulateDataBaseAccess(userId);
+                        CompletableFuture.runAsync(() -> cacheFetchedData(userId, userData));
+                        return userData;
+                    } catch (InterruptedException e) {
+                        throw new RequestProcessingException("Failed to process "
+                                + "the request for userId=" + userId, e);
+                    }
+                });
```

Comment:
Be cautious with lambdas that throw checked exceptions inside `CompletableFuture.supplyAsync`. It is a good practice to handle these exceptions inside the lambda to avoid wrapping them with a CompletionException.